sustaingym/envs/building/stochastic_uncontrollable_generator.py

```diff
@@ -0,0 +1,222 @@
+                Winter ambient features.
+
+        Raises:
+            ValueError if no observation_data is given and the class instance
+                has no observations stored.
+        """
+        if observation_data is None and len(self.observations) == 0:
+            raise ValueError("`observation_data` cannot be None")
+        
+        self.summer_observations = []
+        self.winter_observations = []
+
+        if self.episodes is not None:
+            for i in range(self.episodes):
+                this_observation_block = observation_data[i]
+                this_n = len(this_observation_block)
+
+                this_first_season = this_observation_block[:this_n//4]
+                this_summer = this_observation_block[this_n//4:3*this_n//4]
+                this_last_season = this_observation_block[3*this_n//4:]
+                this_winter = np.vstack((this_first_season, this_last_season))
+
+                if i == 0:
+                    self.summer_observations = this_summer
+                    self.winter_observations = this_winter
+                else:
+                    self.summer_observations = np.vstack((self.summer_observations, this_summer))
+                    self.winter_observations = np.vstack((self.winter_observations, this_winter))
+        else:
+            num_observations = observation_data.shape[0]
+            self.summer_observations = observation_data[num_observations//4:3*num_observations//4, :]
+            self.winter_observations = np.vstack((observation_data[:num_observations//4, :], observation_data[3*num_observations//4:, :]))
+
+        # (len_of_season x num_episodes, dim_of_obs_vector)
+        self.summer_observations = np.array(self.summer_observations)
+        self.winter_observations = np.array(self.winter_observations)
+
+        return self.summer_observations, self.winter_observations
+
+    def get_nearest_block_size(self, obs_length: int, block_size: int):
+        """
+        Finds nearest block size to perfectly divide the length of the observation data.
+
+        Args:
+            obs_length: The number of observation vectors in the data.
+            block_size: The desired block size.
+
+        Returns:
+            (one of) low, high: The nearest block size to perfectly divide the length
+                of the data.
+        """
+        low = block_size
+        high = block_size
+        while obs_length%low != 0 and obs_length%high != 0:
+            low -= 1
+            high += 1
+            if low <= 1:
+                low = 2
+        if obs_length%low == 0: return low
+        elif obs_length%high == 0: return high
+
+    def get_empirical_dist(self, 
+                           season: Optional[str] = None, 
+                           this_season_observations: Optional[np.array] = None, 
+                           block_size_on_split: int = 100) -> scipy.stats.rv_continuous:
```

Comment:
1. Please update the typing annotations to use `str | None` instead of `Optional[str]`.
2. It's `np.ndarray` not `np.array`
3. The return type should be `list[scipy.stats.rv_continuous]`, right?